### PR TITLE
upgrade to FreeRTOS V10.2.1

### PR DESCRIPTION
**Pull request by drewfish**
Here's an attempt. I first did a diff between `master` and FreeRTOS V8.2.3 to see what custom changes where done in this library, and tried to preserve them. I also tried to mark them with "Arduino framework integration" (though not all places).

Hopefully this addresses #2.

### CODE REVIEW

**Comment by BriscoeTech** [comment]
Hello @drewfish,

Thanks for the request! I will make a development branch for this sometime soon, been looking forward to implementing the newer version of the rtos :-)

**Comment by drewfish** [comment]
Cool, no hurries. I've been tracking the other PRs and can keep this PR updated with other changes that land on the master branch.

**Comment by BriscoeTech** [comment]
Hello @drewfish 

Well... I made a mistake lol. I assigned your pr to a new development branch, and then because there was conflicts realized I have to use the command line to checkout your pr. I didn't need the new branch after all since I can check your pr out directly apparently using the command line, so I deleted the branch. Seems harmless right? Github decided to close this pr since the assigned brach no longer existed, and I cannot undo or remap the request back to the original master branch now.

So long story short, could you repost your pr request. Sorry about that!

Thanks again, I will get it right next time ;-)

**Comment by drewfish** [comment]
No worries, I actually expected to redo this PR after the other changes landed on master.
